addons/stock/models/stock_quant.py

```diff
@@ -18,9 +18,40 @@ class StockQuant(models.Model):
     _description = 'Quants'
     _rec_name = 'product_id'
 
+    def _default_location_id(self):
+        if self.env.context.get('active_model') == 'product.template' or self.env.context.get('active_model') == 'product.product':
+            if not self.env['res.users'].has_group('stock.group_stock_multi_locations'):
+                company_user = self.env.user.company_id
+                warehouse = self.env['stock.warehouse'].search([('company_id', '=', company_user.id)], limit=1)
+                if warehouse:
+                    return warehouse.lot_stock_id.id
+                else:
+                    raise UserError(_('You must define a warehouse for the company: %s.') % (company_user.name,))
```

Comment:
i guess raising here is useless if it wasn't part of the specs, just don't set a default

Reply:
When you come on the quant view from the product 'On Hand' stat button, the field `location_id` is invisible if multilocation is inactive. So we need to set a default location as `location_id` is required to create a quant.
Maybe I raise the error only when multilocation is inactive ?